map/chart_generator.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "indexer/feature_altitude.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/deque.hpp"
+#include "std/vector.hpp"
+
+/// \brief fills uniformAltitudeDataM with altitude data which evenly distributed by
+/// |resultPointCount| points. |distanceDataM| and |altitudeDataM| form a curve of route altitude.
+/// This method is used to generalize and evenly distribute points of the chart.
+void NormalizeChartData(deque<double> const & distanceDataM,
```

Comment:
>  What's the problem to insert an element in a vector?

No problem. It's just linear to the vector size.

> Or, even better, why you can't do following:

It's also a possible solution. But in that case the functions RoutingSession::GetSegDistanceM(), Route::GetSegDistanceM(), FollowedPolyline::GetSegDistanceM() would have a strange meaning in my point of view. They weren't just getters but "adders".

What do you think about that?